subsys/bluetooth/controller/ll_sw/ll_master.c

```diff
@@ -36,6 +36,10 @@ u32_t ll_create_connection(u16_t scan_interval, u16_t scan_window,
 
 #if defined(CONFIG_BLUETOOTH_CONTROLLER_PRIVACY)
 	ll_filters_scan_update(filter_policy);
+	if (own_addr_type >= BT_ADDR_LE_PUBLIC_ID) {
```

Comment:
I think it would be safer (future-proof) and clearer for the reader if you compared against the (two, I think?) explicit values that you're looking for here, instead of assuming a specific ordering of the values. Same goes for other places where you test for address type values.